### PR TITLE
This is a movie and tv show app

### PR DESCRIPTION
**Pull request by richie1988**
### Collaboration Team members
- [A ] Richard sikaonga
- [B ] Ismail Halliru Muhammad
### Highlights what we implemented on this project and what we did to come up with this project
TVMaze and Involvement APIs were used to render, send, and receive data	A Single Page App was built using mainly JavaScript	Comment and Reservation Popups were activated	Dynamic likes were displayed	ES6 modules were used	Webpack bundling was utilized	Tests were also 
- Used ES6 syntax.
- we implemented  Send and receive data from an API.
### demo highlights
[https://www.loom.com/share/8489447b36fb4be0ae7e317ae1e9e78e?sid=bed2b077-e524-453d-b1cf-44b2b071142f](url)

### CODE REVIEW

File: src/module/homepagedisplayer.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    button.addEventListener('click', async (e) => {
+      const likeID = e.target.id;
+      const likeCount = e.target.nextElementSibling;
+      if (e.target.classList.contains('like')) {
+        const res = await addLike(likeID);
+        const value = parseInt(likeCount.textContent, 10);
+        e.target.classList.add('like2');
+        e.target.classList.remove('like');
+        if (res.status === 201) {
+          likeCount.innerHTML = value + 1;
```
Comment (FrederickMih) [inline src/module/homepagedisplayer.js]: - [ ] Hi team, based on the requirements of this project, each counter should be implemented as a separate module. We are expected, to still create a specific function to calculate counter numbers on each page :+1: Let's kindly fix this.
- [ ] The counters functions should get covered with unit tests using [the Jest Framwework](https://jestjs.io/) and kindly make sure to cover all the edge cases.

File: README.md [hunk 2]
```diff
@@ -1 +1,184 @@
...
+- **[More dynamic styling]**
+- **[Improved UX interaction]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/features/issues).
```
Comment (FrederickMih) [inline README.md]: - [ ] Let's kindly provide the correct issue link of our project. This allow someone to easily raised an issue and make contributions just my clicking on the correct link. This will equally render our readme file professional :+1:

File: src/module/homepagedisplayer.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    list.appendChild(commentBTN);
+
+    contentContainer.appendChild(list);
+  });
+  const listItems = contentContainer.querySelectorAll('li');
+  const menuList = document.querySelectorAll('.nav-link');
+  const likeEle = document.querySelectorAll('.like');
+  const openPopButtons = document.querySelectorAll('.comment-button');
+  const listCounter = document.createElement('span');
+  listCounter.textContent = `(${listItems.length})`;
```
Comment (FrederickMih) [inline src/module/homepagedisplayer.js:73]: - [ ] Hi team, .length  would actually counts all the moves :+1: unfortunately we were told to use a distinct function for this specific in a different module. Kindly adhere to the requirements of this project and fix :wrench: this :+1: 
- [ ] Please also remember to test the counter function using jest when you're done implementing :+1:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @richie1988 @yayaismaail , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Great app design. Look & Feel so appealing to me :rocket: 
- Great job dynamically creating must of your elements. I think this is a great way of learning :+1: 
- Webpack is correctly set up and use to bundle all files :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/module/homepagedisplayer.js (hunk 3 above)
Comment (richie1988) [reply src/module/homepagedisplayer.js:73]: the length propert its for the screen which I when you just load the screen

File: src/module/likescounter.js
Comment (shahadat3669) [inline src/module/likescounter.js:1]: - [ ] Great job on improving your code so far! It's impressive to see your dedication to enhancing your work. However, there are still some areas that require attention. As suggested by the previous reviewer, you've added a separate module for tracking likes count. As the previous reviewer mentioned, to ensure the functionality works as intended, it's necessary to test the counters functions using the Jest Framework. Please make sure that all edge cases are accounted for and covered with unit tests. 🙏

File: README.md [hunk 4]
```diff
@@ -1 +1,184 @@
...
+- **[More dynamic styling]**
+- **[Improved UX interaction]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://docs.github.com/en/issues/tracking-your-work-with-issues/quickstart).
```
Comment (shahadat3669) [inline README.md]: - [ ] As per the previous reviewer's request, please add the issue link of your current project. This will make it easier for users to contact you. Currently, you are redirecting to the github documentation about how the issues work. Kindly replace it with your current Repo issue link. 🙏

File: src/module/homepagedisplayer.js
Comment (shahadat3669) [inline src/module/homepagedisplayer.js:1]: Hi there, 👋 I noticed that the movies counter is no longer present on your homepage. Just wanted to clarify that the previous reviewer's comments were not related to the removal of the movie counter, but rather focused on improving the code structure. He suggested creating separate module files for each counter and adding unit tests to ensure proper functionality.
- [ ] Therefore, I kindly request that you take some time to review each comment and update your code accordingly. It's recommended to add separate module files for each counter (movies, likes, comments) and include separate test files for each counter. This will help organize your code, make it easier to maintain, and reduce the chance of errors. 🙏  Thank you for your attention to this matter.

File: src/module/commenthandler.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        username: userName,
+        comment,
+      }),
+    });
+    if (!response.ok) throw new Error(`Error creating comment for id ${id}`);
+    return response;
+  } catch (error) {
+    return error;
+  }
+};
```
Comment (shahadat3669) [inline src/module/commenthandler.js:51]: - [ ] Great job on updating the code! However, it looks like there might be a new bug that needs to be fixed. Before the update, users were able to successfully add new comments. Now, there seems to be an issue where the item_id becomes undefined, preventing users from adding comments with the movie's ID. It would be great if you could take a look and address this error. Thank you! 👋 

[Capston-Kanban-projects.webm](https://github.com/richie1988/KanbanCapstone/assets/55840999/3b8c100e-09b1-4b85-9981-fa2410240160)

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @richie1988, @yayaismaail, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ The web app uses an API
✅ API allows for getting a list of items with a unique item ID
✅ use the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) to record the different user interactions (likes, comments and reservations)


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


You can also consider:
- [Optional] Upon reviewing your project, I noticed that the base branch name on your Github page is set as `main`. However, according to [GitFlow documentation](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md#:~:text=Once%20you%20start%20working%20on%20any%20project%2C%20initialize%20its%20main%20branch.%20You%20also%20need%20to%20create%20a%20development%20branch.%20Then%20you%20need%20to%20make%20your%20development%20branch%20the%20default%20branch.), the base branch name should be `development` for the GitHub repository. I kindly request you update the base branch name by following the provided documentation.

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Strangeal — APPROVED** [review]
Hi @richie1988, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Added project link in `README` ✅ 
- Added counters in separate files ✅ 
- Fixed comment `bug`🐛 ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
